### ISSUE TITLE
[feat] Create API to get Linux distribution name

### ISSUE THREAD

**Issue by astudentinearth** [tauri-apps/tauri#6803]
### Describe the problem

When using the current `os` APIs, we can get the operating system name, but the returned name is limited to "Linux", "Windows" and "Darwin". However, there are a lot of different Linux distributions to choose from. Exposing an easy to use API to get the Linux distribution name/version would be helpful in some circumstances, including but not limited to:
- Creating more detailed crash reports, which could now include the Linux distribution name and version, making it easier for the developers to handle such problems
- Allowing applications to show a section where the user can copy such system information, and directly paste to bug reports.
- Handle rare distribution specific cases - if any - when developing apps specifically for the Linux platform

### Describe the solution you'd like

A minimally viable implementation of such API.
Solution is not very complicated to implement. Extracting the `NAME` or `PRETTY_NAME` keys in the `/etc/os-release` file is enough to achieve this in almost every Linux distribution. Then we could create an API method like `getDistributionName()`, which returns either a string containing the distribution name or null in case the method was invoked on a non-Linux platform. Throwing an error could be considered as well.

**Comment by amrbashir** [comment]
Can't you just use `shell.Command` API to execute `uname` and parse it?
```js
let output = await new Command("uname", "-a").execute();
// parse output.stdout for the distro name
```

Do you know if electron or deno or nodejs have a similar API?

**Comment by astudentinearth** [comment]
I just checked the frameworks you mentioned, and they don't have such API. Found a couple of people who asked for such method, but they were usually told to just use the shell command you mentioned. The `shell` API can restrict what commands can be used, meaning this doesn't create an attack vector if configured correctly. 
The `uname -a` command will put the OS name to the end, right after hardware platform, meaning theoretically one could determine the hardware platform first, find the last occurrence of it in the output and substring from that position.
My concern was about security when enabling shell access, but turns out the allowlist configuration is much more strict about this.
After writing this comment, my proposal feels unnecessary, I hope I didn't waste your time.
I believe we can close this issue, as your answer already resolves the problem of anyone who could ask about it.
Wish you a great day 🙂
